Add prop-types

carlosesilva opened the pull request:
This PR installs the prop-types package and defines the proptypes for all the components.

This PR also changes the Question component's prop signature from having a single questionData prop to exposing all of the previous questionData properties as individual props.
This allows us to have more flexibility when defining the PropTypes in a separate file and sharing them between different components.

Note that this change in component prop signature is a breaking change and we will need to specify that in the changelog.

jdub233 commented:
This looks great.  One thing I'm wondering, are some of these proType shapes the same from component to component?  Is there a way to share a common definition across the components?

carlosesilva commented:
That is is an interesting question. I could see that being very beneficial so I am assuming someone out there has already done something similar? I will take a look to see if I find anything

carlosesilva commented:
I extracted the proptype definitions to a separate file so that we can share them across components. I have also updated the description of this PR to add more information about the changes.
